src/Makefile.am

```diff
@@ -32,6 +33,22 @@ snap_confine_CFLAGS += $(APPARMOR_CFLAGS) $(SECCOMP_CFLAGS) $(LIBUDEV_CFLAGS)
 snap_confine_LDADD += $(APPARMOR_LIBS) $(SECCOMP_LIBS) $(LIBUDEV_LIBS)
 endif
 
+sanity_test_SOURCES = \
+	sanity.c \
+	sanity.h \
+	$(snap_confine_SOURCES)
+# Comple sanity-tests from the same sources as snap-confine but
+# define the _SANITY_TESTING macro so that test code can be
+# included. This lets us put tests right next to actual static
+# functions for better visiblitiy.
+sanity_test_CFLAGS = $(snap_confine_CFLAGS)
```

Comment:
You could write `sanity_test_CFLAGS = $(snap_confine_CFLAGS)  $(GLIB_CFLAGS)` and avoid the extra lines below.